### PR TITLE
Fix categorical transform

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
An error happens when the loaded categorical features include both integers and strings. This may happen when loading features from CSV files.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/transform.py [hunk 1]
```diff
@@ -379,7 +379,7 @@ def pre_process(self, feats):
 
         feats = feats[feats != None] # pylint: disable=singleton-comparison
         if self._separator is None:
-            return {self.feat_name: np.unique(feats)}
+            return {self.feat_name: np.unique(feats.astype(str))}
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/transform.py:382]: is it good to convert everything to strings?

**Comment by zheng-da** [comment]
you need to maintain the backward compatibility

**Comment by classicsong** [comment]
> you need to maintain the backward compatibility

I have added.

File: python/graphstorm/gconstruct/transform.py [hunk 2]
```diff
@@ -351,7 +351,8 @@ class CategoricalTransform(TwoPhaseFeatTransform):
     def __init__(self, col_name, feat_name, separator=None, transform_conf=None):
         self._val_dict = {}
         if transform_conf is not None and 'mapping' in transform_conf:
-            self._val_dict = transform_conf['mapping']
+            self._val_dict = \
+                {str(key): val for key, val in transform_conf['mapping'].items()}
```
Comment (classicsong) [inline python/graphstorm/gconstruct/transform.py:355]: @zheng-da fix the backward compatibility

Comment (zheng-da) [reply python/graphstorm/gconstruct/transform.py:355]: you should add comments here to explain why the keys are converted to strings.

**Review by zheng-da — APPROVED** [review]
please add comments to explain the data type conversion. otherwise, it looks good to me.

**Pull request merged**